iree/compiler/Conversion/Common/VectorTransferOptimization.cpp

```diff
@@ -22,7 +24,45 @@ namespace {
 
 struct VectorTransferOptimizationPass
     : public PassWrapper<VectorTransferOptimizationPass, FunctionPass> {
-  void runOnFunction() override { vector::transferOpflowOpt(getFunction()); }
+  void runOnFunction() override {
+    vector::transferOpflowOpt(getFunction());
+    // Delete potential dead alloc and associated ops after store to load
+    // forwarding.
+    eraseDeadAllocAndStores(getFunction());
+  }
+
+ private:
+  // Track temporary allocations that are never read from. If this is the case
+  // it means both the allocations and associated stores can be removed.
+  void eraseDeadAllocAndStores(FuncOp funcOp) {
```

Comment:
can we move these out to be static methods in the file. I dont think they need to be class members.

Reply:
Good point, done.